scene/3d/soft_dynamic_body_3d.cpp

```diff
@@ -88,10 +88,10 @@ void SoftDynamicBodyRenderingServerHandler::set_normal(int p_vertex_id, const vo
 	memcpy(&n, p_vector3, sizeof(Vector3));
 	n *= Vector3(0.5, 0.5, 0.5);
 	n += Vector3(0.5, 0.5, 0.5);
+	Vector2 res = n.octahedron_encode();
 	uint32_t value = 0;
-	value |= CLAMP(int(n.x * 1023.0), 0, 1023);
-	value |= CLAMP(int(n.y * 1023.0), 0, 1023) << 10;
-	value |= CLAMP(int(n.z * 1023.0), 0, 1023) << 20;
+	value |= (uint16_t)CLAMP(res.x * 65535, 0, 65535);
```

Comment:
I know the code surrounding these changes uses these numbers directly; but would it be preferred to change these "magic numbers" into enums (for example stdint.h has enums for UINT16_MAX) for readability or should we keep this implementation as is?

Reply:
I guess you can do constexp and pow 2 to generate these.

Reply:
I would leave as-is. It is quite obvious from context what you are doing.